### PR TITLE
[BACK-2679] Resolve dependabot issues

### PR DESCRIPTION
**Pull request by darinkrauss**
- Resolve dependabot issues
- Update to node v20.11.0
- Remove request dependency
- Add lint via jshint
- Add .nvmrc
- Update .travis.yml
- Update Dockerfile
- Bump version
- https://tidepool.atlassian.net/browse/BACK-2679

### CODE REVIEW

File: .nvmrc [hunk 1]
```diff
@@ -0,0 +1 @@
+v20.11.0
```
Comment (jh-bate) [inline .nvmrc:1]: version ? `v20.10.0`

**Review by jh-bate — APPROVED** [review]

File: .nvmrc (hunk 1 above)
Comment (darinkrauss) [reply .nvmrc:1]: Updated PR description.

File: test/integration/messagesService_integration_tests.js [hunk 2]
```diff
@@ -23,7 +25,7 @@ var sinon = salinity.sinon;
 var mockableObject = salinity.mockableObject;
 
 var env = {
-  httpPort: 21000,
+  httpPort: 21003,
```
Comment (lostlevels) [inline test/integration/messagesService_integration_tests.js:28]: Is there a reason this changed? I'm guessing it doesn't matter to much for a test.

**Review by lostlevels — APPROVED** [review]

File: test/integration/messagesService_integration_tests.js (hunk 2 above)
Comment (darinkrauss) [reply test/integration/messagesService_integration_tests.js:28]: It clashed with a test in another repo.

**Pull request merged**